### PR TITLE
Scenes

### PR DESCRIPTION
**Pull request by riknoll**
Adding a stack for the game state on top of Peli's stack for event handlers

### CODE REVIEW

File: libs/game/game.ts [hunk 1]
```diff
@@ -14,11 +71,13 @@ namespace game {
     export let debug = false;
     export let flags: number = 0;
```
Comment (pelikhan) [inline libs/game/game.ts]: Migrate this one too

**Comment by pelikhan** [comment]
The keys should just work since they are compatible with control.eventContext.

File: libs/game/game.ts [hunk 2]
```diff
@@ -14,11 +71,13 @@ namespace game {
...
-    let __isOver = false
+
+    export let scene: Scene;
+    let sceneStack: Scene[];
+
+
     let __waitAnyKey: () => void
-    let __background: Background;
-    let __tileMap: tiles.TileMap;
+    let __isOver = false;
```
Comment (mmoskal) [inline libs/game/game.ts]: I might have started it, but I'm not sure why we have all these underscores here...

**Comment by pelikhan** [comment]
The Monaco editor hides members stating with __ in inyellisense. Not useful if not exported.

________________________________

**Pull request merged**